src/prefs.ts

```diff
@@ -43,6 +46,11 @@ class RandomWallpaperSettings extends ExtensionPreferences {
         window.set_search_enabled(true);
 
         backendConnection.setBoolean('pause-timer', true);
+        // save and observe the pause-timer state while settings are open
+        this.pauseTimerState = backendConnection.getBoolean('pause-timer');
+        backendConnection.observe('pause-timer', () => {
+            this.pauseTimerState = backendConnection.getBoolean('pause-timer');
+        });
```

Comment:
> With this change it's always pausing the timer because of the ordering.

Uff :facepalm: thanks!

> My original intention was that the timer doesn't fire while changes are being made.

That makes kind of sense to me, but I am not sure how important or confusing this functionality is in the end. I'd just leave it in for now. We can remove it at any time if we decide to.

> If we're going further with this idea, we ideally would disable the pause switch while the pref menu is open to prevent switching the timer on with the menu open. This also would remove the need to observe the current state.

I'm not really convinced that this is better from a user perspective without some clear and visible documentation on this behavior. I think I'd rather remove this functionality entirely if we run into any problems down the road.